packages/backend-data/src/factory.ts

```diff
@@ -132,38 +132,12 @@ class DataGenerator implements ConstructContainerEntryGenerator {
       );
     }
 
-    let authorizationModes;
-
-    /**
-     * TODO - remove this after the data construct does work to remove the need for allow-listed IAM roles
-     */
-    const functionSchemaAccessRoles = functionSchemaAccess.map(
-      (accessEntry) =>
-        accessEntry.resourceProvider.getInstance(this.getInstanceProps)
-          .resources.lambda.role!
+    const authorizationModes = convertAuthorizationModesToCDK(
+      this.getInstanceProps,
+      this.providedAuthConfig,
+      this.props.authorizationModes
     );
 
-    try {
-      authorizationModes = convertAuthorizationModesToCDK(
-        this.getInstanceProps,
-        this.providedAuthConfig,
-        this.props.authorizationModes,
-        functionSchemaAccessRoles
-      );
-    } catch (error) {
-      throw new AmplifyUserError<AmplifyDataError>(
-        'InvalidSchemaAuthError',
-        {
-          message:
-            error instanceof Error
-              ? error.message
-              : 'Failed to parse authorization modes.',
-          resolution: 'Ensure the auth rules on your schema are valid.',
-        },
-        error instanceof Error ? error : undefined
-      );
-    }
```

Comment:
I removed this try-catch as it didn't make much sense. 
The `convertAuthorizationModesToCDK` doesn't seem to access schema at all.